keras_retinanet/models/mobilenet.py

```diff
@@ -0,0 +1,56 @@
+import keras
+from keras import backend as K
+from keras.models import Model
+from keras.applications.mobilenet import MobileNet, BASE_WEIGHT_PATH, get_file
+
+from ..models import retinanet
+
+custom_objects = retinanet.custom_objects.copy()
+
+
+def download_imagenet(backbone):
+    alpha = 1.0
+    rows = backbone
+
+    # load weights
+    if K.image_data_format() == 'channels_first':
+        raise ValueError('Weights for "channels_last" format '
+                         'are not available.')
+    if alpha == 1.0:
+        alpha_text = '1_0'
+    elif alpha == 0.75:
+        alpha_text = '7_5'
+    elif alpha == 0.50:
+        alpha_text = '5_0'
+    else:
+        alpha_text = '2_5'
+
+    model_name = 'mobilenet_%s_%d_tf_no_top.h5' % (alpha_text, rows)
+    weigh_path = BASE_WEIGHT_PATH + model_name
+    weights_path = get_file(model_name,
+                            weigh_path,
+                            cache_subdir='models')
+
+    return weights_path
+
+
+def mobilenet_retinanet(num_classes, backbone=224, inputs=None, **kwargs):
```

Comment:
`backbone` is not used?

Reply:
The parameters of this network (when loading imagenet) are:
        alpha: controls the width of the network.
            - If `alpha` < 1.0, proportionally decreases the number
                of filters in each layer.
            - If `alpha` > 1.0, proportionally increases the number
                of filters in each layer.
            - If `alpha` = 1, default number of filters from the paper
                 are used at each layer.
            - If imagenet weights are being loaded, alpha can be one of' `0.25`, `0.50`, `0.75` or `1.0` only.'
      depth_multiplier 
          -if imagenet weights are being loaded, 'depth multiplier must be 1.
      input_shape: If imagenet weights are being loaded, '
                             'input must have a static square shape (one of '
                             '(128,128), (160,160), (192,192), or (224, 224))

So they are different from the resnet.

Reply:
As they are different from the resnet we should refactor the backbone loading in the train.py script. I already started, but I prefer to let you finish this. This issue will appear again when i'll do the next pull request with the Densenet backbone.

Reply:
Yeah, we should probably have some better way to configure a backbone including more detailed configuration parameters. We might have to rework command line parsing to really support that nicely.